tests/lib/eslint/eslint.js

```diff
@@ -143,8848 +147,9092 @@ describe("ESLint", () => {
+                        languageOptions: {
+                            globals: {
+                                window: false
+                            }
+                        },
+                        rules: {
+                            "no-alert": 0,
+                            "no-undef": 2
+                        }
                     }
                 });
-                const results = await eslint.lintFiles([fs.realpathSync(getFixturePath("rules", "test", "test-custom-rule.js"))]);
+                const results = await eslint.lintFiles([fs.realpathSync(getFixturePath("globals-browser.js"))]);
 
                 assert.strictEqual(results.length, 1);
-                assert.strictEqual(results[0].messages.length, 2);
-                assert.strictEqual(results[0].messages[0].ruleId, "test/example-rule");
+                assert.strictEqual(results[0].messages.length, 0);
                 assert.strictEqual(results[0].suppressedMessages.length, 0);
-
             });
-        });
 
-        describe("cache", () => {
+            it("should return zero messages when given a config with sourceType set to commonjs and Node.js globals", async () => {
+                eslint = new ESLint({
+                    flags,
+                    cwd: path.join(fixtureDir, ".."),
+                    overrideConfigFile: getFixturePath("configurations", "env-node.js")
+                });
+                const results = await eslint.lintFiles([fs.realpathSync(getFixturePath("globals-node.js"))]);
 
-            /**
-             * helper method to delete a file without caring about exceptions
-             * @param {string} filePath The file path
-             * @returns {void}
-             */
-            function doDelete(filePath) {
-                try {
-                    fs.unlinkSync(filePath);
-                } catch {
+                assert.strictEqual(results.length, 1);
+                assert.strictEqual(results[0].messages.length, 0, "Should have no messages.");
+                assert.strictEqual(results[0].suppressedMessages.length, 0);
+            });
 
-                    /*
-                     * we don't care if the file didn't exist, since our
-                     * intention was to remove the file
-                     */
-                }
-            }
+            it("should not return results from previous call when calling more than once", async () => {
+                eslint = new ESLint({
+                    flags,
+                    cwd: path.join(fixtureDir, ".."),
+                    overrideConfigFile: getFixturePath("eslint.config.js"),
+                    ignore: false,
+                    overrideConfig: {
+                        rules: {
+                            semi: 2
+                        }
+                    }
+                });
+                const failFilePath = fs.realpathSync(ge
```

Comment:
I don't think we explicitly defined this behavior. Technically, throwing an error is correct because our logic says "for a given file, if you can't find a config file that tells you what to do with it, throw an error."

I'm not sure how common it would be for someone to run `eslint .` in a directory without a config file. This might be a case where we leave this behavior as-is and see what feedback we get. Otherwise, I'd be back to checking "if pattern is `.` and file is an immediate child of `.`, then ignore it".